src/main/scala/com/mozilla/telemetry/heka/package.scala

```diff
@@ -7,13 +7,63 @@
 package com.mozilla.telemetry
 
 import com.google.protobuf.ByteString
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.JsonDSL._
+import scala.util.Try
 
 package object heka {
   class RichMessage(m: Message) {
+
+    /** Deserialize the Heka message as a native Scala Map. */
     def fieldsAsMap: Map[String, Any] = {
       val fields = m.fields
       Map(fields.map(_.name).zip(fields.map(field)): _*)
     }
+
+    /** Reconstructs a Heka message as a json4s JValue.
+      *
+      * The message is parsed and serialized into JSON. Keys that have been extracted
+      * into the toplevel document are inserted into to the proper place in the document.
+      *
+      * @return reconstructed JValue
+      */
+    def toJValue: JValue = {
+      val fields = Map(m.fields.map(_.name).zip(m.fields.map(fieldAsJValue)): _*)
+
+      lazy val submission = fields.getOrElse("submission", JNothing)
+      val payload = m.payload match {
+        case Some(payload: String) => {
+          val json = Try(parse(payload)).getOrElse(JNothing)
```

Comment:
The code for main summary can match on JValue or JNothing to determine whether the submission is empty or invalid.

Reply:
This isn't just about the changes we'll make to main summary -- it's for any future users that aren't involved in this change. Would they know they should check for empty submission values? I'd venture no -- it's much safer to throw the exception and let the future developer handle the exception when it happens as opposed to discovering this later once they realize they have unexpectedly empty rows.

Alternately, idiomatically in scala you could signal to the future developer that they should check for empty values by making the return value an option. Using empty values instead of None are strongly frowned otherwise.

Reply:
I see, JValue does seem a bit opaque. I'll play around with Option and Try now that I have the relevant test cases.